Source/Core/DolphinTool/HeaderCommand.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "DolphinTool/HeaderCommand.h"
+#include "DiscIO/Blob.h"
+#include "DiscIO/Volume.h"
+#include "DiscIO/VolumeDisc.h"
+
+#include <OptionParser.h>
+#include <optional>
+
+namespace DolphinTool
+{
+int HeaderCommand::Main(const std::vector<std::string>& args)
+{
+  auto parser = std::make_unique<optparse::OptionParser>();
+
+  parser->usage("usage: header [options]...");
+
+  parser->add_option("-i", "--input")
+      .type("string")
+      .action("store")
+      .help("Path to disc image FILE.")
+      .metavar("FILE");
+
+  parser->add_option("-b", "--block_size")
+      .action("store_true")
+      .help("Optional. Print the block size of GCZ/WIA/RVZ formats, then exit.");
+
+  parser->add_option("-c", "--compression")
+      .action("store_true")
+      .help("Optional. Print the compression method of GCZ/WIA/RVZ formats, then exit.");
+
+  parser->add_option("-l", "--compression_level")
+      .action("store_true")
+      .help("Optional. Print the level of compression for WIA/RVZ formats, then exit.");
+
+  const optparse::Values& options = parser->parse_args(args);
+
+  // Validate options
+  const std::string input_file_path = static_cast<const char*>(options.get("input"));
+  if (input_file_path.empty())
+  {
+    std::cerr << "Error: No input set" << std::endl;
+    return 1;
+  }
+
+  bool enable_block_size = options.is_set_by_user("block_size");
+  bool enable_compression_method = options.is_set_by_user("compression");
+  bool enable_compression_level = options.is_set_by_user("compression_level");
+
+  // Open the volume + blob reader, plus get blob type
+  std::shared_ptr<DiscIO::VolumeDisc> volume = DiscIO::CreateDisc(input_file_path);
```

Comment:
You're not actually using the volume for anything other than getting the blob reader, so I would suggest just creating a blob reader directly.